### ISSUE TITLE
Error when parsing f-strings with multiple arguments and formatting options

### ISSUE THREAD

**Issue by mallamanis** [davidhalter/parso#51]
Thanks for the very useful library! I believe that there is an issue with parsing multiple arguments of an f-string when one has a formatting argument:

```[python]
import parso

simple = "print(f'Some {x} and some {y}')"
grammar = parso.load_grammar()
module = grammar.parse(simple)
err = grammar.iter_errors(module)
print(f'`simple` f-string: {len(err)}')

with_formatting = "print(f'Some {x:.2f} and some {y}')"
grammar = parso.load_grammar()
module = grammar.parse(with_formatting)
(err,) = grammar.iter_errors(module)
print(f'`with_formatting` f-string: {err.message}')

with_formatting2 = "print(f'Some {x:,} and some {y}')"
grammar = parso.load_grammar()
module = grammar.parse(with_formatting2)
(err,) = grammar.iter_errors(module)
print(f'`with_formatting2` f-string: {err.message}')

single = "print(f'Some {x:,}')"
grammar = parso.load_grammar()
module = grammar.parse(single)
err = grammar.iter_errors(module)
print(f'`single` f-string: {len(err)}')
```

prints

```
`simple` f-string: 0
`with_formatting` f-string: SyntaxError: invalid syntax
`with_formatting2` f-string: SyntaxError: invalid syntax
`single` f-string: 0
```

**Comment by davidhalter** [comment]
Should be good now. Release coming soon, just trying to fix a few last things in the diff parser.

I would be very interested in knowing how you/or your company are using `iter_errors`, because I haven't seen a lot of usages yet (even though it's quite a good feature).

**Comment by mallamanis** [comment]
Thanks a lot for fixing this!

I used parso for a [ml4code](http://ml4code.github.io/) research project with a PhD student I am collaborating with. Nothing big/fancy. As I was parsing in batch a large corpora of code, I needed to see where/why parsing failed and that's where I started using `iter_errors` (and uncovered this issue). Unfortunately, I haven't seen any other usages of `iter_errors` elsewhere, beyond the documentation examples...

Thanks a lot again!
